agent/loop.py

```
@@ -511,8 +574,244 @@ async def _dispatch_tool(
+            state = {
+                "session_id": session_id or "unknown",
+                "status": "running",
+                "last_updated": time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime()),
+                "goal": plan.goal,
+                "step_count": len(plan.steps),
+                "risks": plan.risks,
+                "requires_risky_review": plan.requires_risky_review,
+            }
+            state_file = state_dir / "agent-state.json"
+            state_file.write_text(json.dumps(state, indent=2), encoding="utf-8")
+        except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)